### DISCUSSION TITLE
What is the way to run external js function in client side in SvelteKit.

### DISCUSSION THREAD

**Discussion by shellking4** [sveltejs/kit#9917] [Q&A]
I have an external js file which declare a function setUpAppBehavior. The function is responsible for some cool animation in my website

First I included the js file in my app.html

```html
<!DOCTYPE html>
<html lang="en">

<head>
	<meta charset="utf-8" />
	<link rel="icon" href="%sveltekit.assets%/favicon.png" />
	<meta name="viewport" content="width=device-width" />
	<meta charset="utf-8">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<link rel="stylesheet" type="text/css" href="/assets/css/style.css" />
	%sveltekit.head%
</head>

<body data-spy="scroll" data-target="#navbar-main">
	<div style="display: contents">
		%sveltekit.body%
	</div>

	<script src="/assets/js/vendor/jquery.min.js"></script>
	<script src="/assets/js/script.js"></script>
</body>

</html>
 ```

And then in my +layout.svelte file I did this 
```ts
<script lang="ts" context="module">
    import { browser } from '$app/environment';
    import Footer from "../libs/components/Footer.svelte";
    import Header from "../libs/components/Header.svelte";
    declare var setUpAppBehavior: any;

    /* Doing this way, the js function got executed  and 
      I have the behavior I want both in development and in production after build... 
      but its throwing undefined for process in browser when I'm in development
    */
    if ((process as any)?.client) setUpAppBehavior();

    /* Doing this way, the js function got executed and I have the behavior 
       I want only in production after build. It's not working in development.
    */
    if (browser) setUpAppBehavior();
    
</script>

<Header />
     <slot />
<Footer />
 ```
I tried also with this snippet ``` let isServer = import.meta.env.SSR;  ``` but same result as in the browser case...

What am I missing ? What this the right way to do this ? Run the js function in client side.

**Comment by cbarber1** [comment]
did you figure this out? I am having the exact same problem at the moment

**Comment by leoj3n** [comment]
I mean using `jquery` on a SvelteKit website seems a bit counter-intuitive to me. As it's in the name, Svelte is kind of trying to get away from bulky libraries like jquery, as I understand it. Regardless of your need to use jquery, you should be able to interoperate with external libraries as long as you understand the Svelte component lifecycle. I'm not sure what your `setUpAppBehavior` function does, but if it doesn't depend on the Svelte lifecycle you should be able to just call it in a `<script>` tag that follows the `jquery` script...

```js
	<script src="/assets/js/vendor/jquery.min.js"></script>
	<script src="/assets/js/script.js"></script> 
	<script>
		setUpAppBehavior();
	</script>
```

That should execute regardless of the development or production environment.

If you do need to wait for the layout or component code in Svelte to have been executed, you can use `onMount`...

- https://svelte.dev/docs/svelte#onmount
- https://www.reddit.com/r/sveltejs/comments/p5p386/trying_to_understand_script_vs_onmount/

I see you are trying to tell typescript that the setUpAppBehavior is coming from the outside with `declare`. I don't know if this is the best, if you can get the code for your animation into Svelte with an import or something that might be better.

**Comment by leoj3n** [comment]
```js
	import { onMount } from 'svelte';

	onMount(() => {
		alert('hello');
	});
```

Note that depending on how your `SSR` / `CSR` is setup for your app this may only run on the initial page load.
